--- Pale-Moon-27.5.1_Release/browser/installer/package-manifest.in
+++ Pale-Moon-27.5.1_Release/browser/installer/package-manifest.in
@@ -687,6 +687,7 @@
 ; All the pref files must be part of base to prevent migration bugs
 @RESPATH@/browser/@PREF_DIR@/palemoon.js
 @RESPATH@/browser/@PREF_DIR@/palemoon-branding.js
+@RESPATH@/browser/@PREF_DIR@/all-gentoo.js
 @RESPATH@/goanna.js
 @RESPATH@/defaults/autoconfig/platform.js
 @RESPATH@/defaults/autoconfig/prefcalls.js
